File: public/src/app/instance/instance-query.ts

export class InstanceQuery {

  constructor(public startDate: string,
              public endDate: string,
              public status: number,
              public groupId: string) {}

  static fromJson(json: any) {
    new InstanceQuery(json.start_date, json.end_date, json.status, json.group_id)
  }

  json() {
    return {
      start_date: this.startDate,
      end_date: this.endDate,
      status: this.status?this.status:null,
      group_id: this.groupId
    }
  }

}

File: app/model/InstanceQuery.scala
package model

import java.sql.Date

import play.api.libs.functional.syntax._
import play.api.libs.json.{Format, JsPath}
import util.Util.DateJsonParser

/**
  * Created by chlr on 7/2/17.
  */

case class InstanceQuery(startDate: Date, endDate: Date, status: Option[Int], groupId: String)

object InstanceQuery {

  implicit val formatter: Format[InstanceQuery] = (
    (JsPath \ "start_date").format[Date] and
    (JsPath \ "end_date").format[Date] and
    (JsPath \ "status").formatNullable[Int] and
    (JsPath \ "group_id").format[String]
  )(InstanceQuery.apply, unlift(InstanceQuery.unapply))
}
